### ISSUE TITLE
Viewing Detailed Prediction Results of Caffe

### ISSUE THREAD

**Issue by res97** [BVLC/caffe#1618]
Using command line caffe, we would like to view detailed prediction results. The results, as of now, only show some probability or output associated with batch. The results are in this form:

I1221 18:28:45.424633 18303 caffe.cpp:169] Batch 3, output = 0
I1221 18:28:45.427116 18303 caffe.cpp:169] Batch 3, output = 1
I1221 18:28:45.429867 18303 caffe.cpp:169] Batch 3, output = 0
I1221 18:28:45.434942 18303 caffe.cpp:169] Batch 3, output = 1
I1221 18:28:45.682468 18303 caffe.cpp:169] Batch 4, loss = 0.496991

However, we want the test results in this form:

Imagename1.jpg, output = 0 or prob = 0.4
Imagename2.jpg, output = 1 or prob = 0.2
Imagename3.jpg, output = 0 or prob = 0.3
Imagename4.jpg, output = 1 or prob = 0.4

If anyone can figure this out, it would be a tremendous help to the community.
